models/ssh_key.go

```diff
@@ -472,6 +472,14 @@ func ListPublicKeys(uid int64) ([]*PublicKey, error) {
 		Find(&keys)
 }
 
+// ListPublicLdapSSHKeys returns a list of synchronized public ldap ssh keys belongs to given user.
+func ListPublicLdapSSHKeys(uid int64) ([]*PublicKey, error) {
```

Comment:
It should not touch ldap keys from other sources other the one being synced, so you should add source I'd as argument and search only these. Otherwise if two ldap sources are added one will delete others ldap keys

Reply:
@lafriks - This is fixed now, please review when possible.